hudi-spark/src/main/java/org/apache/hudi/keygen/parser/HoodieDateTimeParserImpl.java

```diff
@@ -95,7 +86,15 @@ public String getOutputDateFormat() {
 
   @Override
   public DateTimeFormatter getInputFormatter() {
-    return this.inputFormatter;
+    TimestampType timestampType = TimestampType.valueOf(config.getString(Config.TIMESTAMP_TYPE_FIELD_PROP));
+    if (timestampType == TimestampType.DATE_STRING || timestampType == TimestampType.MIXED) {
+      DataSourceUtils.checkRequiredProperties(config,
+          Collections.singletonList(Config.TIMESTAMP_INPUT_DATE_FORMAT_PROP));
+      this.configInputDateFormatList = config.getString(Config.TIMESTAMP_INPUT_DATE_FORMAT_PROP, "");
+      return getInputDateFormatter();
+    }
+
+    return null;
```

Comment:
throw an exception?  returning null is like playing with fire :)

Reply:
Taken care of this :)